### DISCUSSION TITLE
Focus on the last space after Yabai restart

### DISCUSSION THREAD

**Discussion by hamadahiro** [asmvik/yabai#2042] [Q&A] [answered]
When I restart Yabai or the computer, the focus always goes to the last space. Why is this happening, and is there a way to force Yabai to remain on the first space?

**Comment by asmvik** [comment]
This has never happened to me. Must be something in your config-file.

**Comment by hamadahiro** [comment]
After more testing, it appears that the issue lies in the fact that it always focuses on the space in which the last app was loaded. I experimented with various combinations of open apps, as well as when all apps are closed. When multiple apps are open, it consistently ends up on a random space. However, when all apps are closed, it always ends up on the first space. To confirm my doubts, I added an app to the Login Items and assigned it to space 3 in yabai. Upon restart, it consistently ends up on space 3.

**Reply by asmvik** [reply]
How do you assign it to space 3 in yabai? yabai can move a window *and* follow focus to that space, or just move the window.

To just move the window: `yabai -m rule --add app="Finder" space=3`
To move the window and follow focus: `yabai -m rule --add app="Finder" space=^3`

**Comment by hamadahiro** [comment]
Here's the code from yabairc where I have assigned Mail app to open in space 3. Everything generally works wonderfully with this setup; it's just this thing that's bugging me from time to time when I have to restart either yabai or the computer and I end up in some random space.

```
yabai -m window --space 1 --focus
yabai -m window --space 2 --focus
yabai -m window --space 3 --focus
yabai -m window --space 4 --focus
yabai -m window --space 5 --focus
yabai -m window --space 6 --focus
yabai -m window --space 7 --focus

# space labels
yabai -m space 1 --label dev
yabai -m space 2 --label web
yabai -m space 3 --label msg
yabai -m space 4 --label org
yabai -m space 5 --label fun
yabai -m space 6 --label art
yabai -m space 7 --label k12 

yabai -m rule --add app="Mail$" 				            	space=msg
```

**Comment by hamadahiro** [comment, marked as answer]
OMG! Just now when I pasted this code, I noticed the problem! I had been looking at this code all the time and never noticed that I had --focus set to every --space! what to say....

Thank you for looking into this anyway, and thanks for yabai. After a few months of using it, I will never look back.
